misc/docerina/src/main/java/org/ballerinalang/docgen/Generator.java

```diff
@@ -130,7 +130,8 @@ public static boolean setModuleFromSyntaxTree(Module module, SyntaxTree syntaxTr
                             hasPublicConstructs = true;
                             Type firstType = Type.fromNode(((UnionTypeDescriptorNode) (typeDefinition.typeDescriptor()))
                                     .leftTypeDesc(), semanticModel);
-                            if (firstType.category.equals("errors")) {
+                            if (firstType.category.equals("errors") ||
+                                    (firstType.category.equals("builtin") && firstType.name.equals("error"))) {
```

Comment:
Is there a possibility to add a unit test for this?

Reply:
Added a unit test